src/page/product-list.page.ts

```diff
@@ -1,16 +1,32 @@
 import {
-  $, ElementFinder, ExpectedConditions, browser,
+  $, $$, ElementArrayFinder, ElementFinder,
+  ExpectedConditions, browser,
 } from 'protractor';
 
 export class ProductListPage {
-  private addToCardMenu: ElementFinder;
+  private addToCardBttn: ElementFinder;
+  private products: ElementArrayFinder;
 
   constructor() {
-    this.addToCardMenu = $('#center_column [title="Add to cart"]');
+    this.products = $$('.product_list.row');
+    this.addToCardBttn = $('#add_to_cart > button');
+
+  }
+
+  private findByProduct(productName: string): ElementFinder {
+    return this.products.filter((product: ElementFinder) =>
+      product.$('[itemprop="name"]').getText().
+      then((text : string) => text === productName)).first();
```

Comment:
What about this:
```
    return this.products
        .filter(
            (product: ElementFinder) =>
                product
                    .$('[itemprop="name"]')
                    .getText()
                    .then((text : string) => text === productName))
        .first();
```

There is no rule of thumb for line breaking, but in general, it is a good practice to start the lines with a method call including the dot `.`